### ISSUE TITLE
[docdb] Master web UI redirect does not respect server_broadcast_addresses

### ISSUE THREAD

**Issue by iSignal** [yugabyte/yugabyte-db#7867]
Repro steps:

1. Set up a cluster with 3 masters where [--server_broadcast_addresse](https://docs.yugabyte.com/latest/reference/configuration/yb-tserver/#server-broadcast-addresses)s for each master differs from --webserver_interface
2. Browse to master UI for a non leader master. Notice it redirects to the leader master using the leader master's webserver_interface address instead of server_broadcast_addresses.

Details:

To identify the address of the leader master from the current master, we use the http addresses specified in the server registration, which is analogous to rpc_bind_addresses but for the web UI. However, while rpc bind addresses can be overriden in specific scenarios using server broadcast addresses, the http addresses cannot be overriden similarly. This causes the master web UI part alone to fail to redirect. 

To see scenarios where server_broadcast_addresses is useful and different from rpc bind addresses, see https://docs.yugabyte.com/latest/reference/configuration/yb-master/#use-private-ip and https://blog.yugabyte.com/yugabyte-db-1-1-new-feature-public-ips-to-simplify-multi-or-hybrid-cloud-database-deployments/. In practice, we hit these in multi cluster k8s deployments or within docker containers also. For example: https://github.com/yugabyte/yugabyte-db/issues/6526

cc @bhavin192

**Comment by druzac** [comment]
I'm aware of two places where we use the address of other masters. 

1. When viewing the status pages for a non-leader master it makes requests to the leader master to get certain information, for example the information populating the /tablet-servers page. 
2. When creating links to the other masters on the root page in the "Masters" table.

[e99e29481e596b53d9261096457829eccc3c253f](https://github.com/yugabyte/yugabyte-db/commit/e99e29481e596b53d9261096457829eccc3c253f) only fixes problem 1. Problem 2 is still open.

**Comment by druzac** [comment]
@bmatican actually, I'm not sure we want identical logic. This is my understanding of the situation:

1. When a non-leader master is making requests for status information to the leader master, the communication is node -> node. So re-using the broadcast IPs vs. private IPs logic [here](https://github.com/yugabyte/yugabyte-db/blob/master/src/yb/common/wire_protocol.cc#L126) seems appropriate. That logic does some munging based on the `use_private_ip` flag and the location of the two nodes. There's an example in [11603](https://github.com/yugabyte/yugabyte-db/issues/11603) of where that logic is needed - one master could print a status page but the other couldn't, because the first master was located in the same zone as the master leader and the second master was located in a different zone.
2. For links, presumably the link will be followed by the browser viewing the status page. I suppose people could proxy requests their browser makes through a server in a cloud zone that could resolve private IPs, but that seems incredibly niche. In general the only IP address of a master that should be used in status pages is the (a?) public, broadcast address. There's no need for any logic to decide between public and private IPs in this case, we should always use the public IP.

What do you think?

**Comment by iSignal** [comment]
Yes agree that the master web UI should show the public IP / server broadcast addresses (essentially we can always pretend we are in a faraway cloud irrespective of the use_private_ip parameter).

**Comment by bmatican** [comment]
@iSignal Would that interfere with Platform's /proxy capabilities? If I recall, there's some form of nginx html re-write that happens, in order to automatically turn all urls into their /proxy versions

Otherwise, I'd agree with @druzac , that since all urls point to an http endpoint, they should probably all use the external facing IP.

**Comment by iSignal** [comment]
@bmatican : Platform only specifies server broadcast addresses in k8s and in this case, docdb autodetects webserver_interface to the local hostname (because the gflag is set to 0.0.0.0), so there should be no real change with this behavior to the master UI in k8s.

But it can help us with some other multi-cluster k8s scenarios which we are working through in OSS k8s so far. If there is some remote case where we need to toggle this behavior on master UI (seems unlikely so far) we can add a http header or CGI param to differentiate them and Platform can set that.
